src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -150,6 +150,25 @@ size_t check_read(FileDesc fd, off_t pos, void* dst, size_t len)
     return ret;
 }
 
+inline void memcpy_tsan_safe(char* dst, char* src, size_t num_bytes)
+{
+#if REALM_SANITIZE_THREAD
+            // Copying an entire page may overwrite arrays which is being read concurrently by
+            // another thread. This is benign because it overwrites data from old version with
+            // the same value as is already there, so the reader sees the correct value. This
+            // is all by design. However, when doing a memcpy of the entire page, TSAN sees
+            // this as a race even though we are replacing concurrently read data with the same
+            // values.
+            for (size_t i = 0; i < num_bytes; ++i) {
+                if (src[i] != dst[i]) {
```

Comment:
Why check here if it should be the same data? Isn't this a contradiction to what is written in the comment?